spec/client/ics-007-tendermint-client/README.md

```diff
@@ -206,44 +206,53 @@ Tendermint client validity checking uses the bisection algorithm described in th
 
 ```typescript
 function verifyClientMessage(
-  clientMsg: ClientMessage) {
+  clientMsg: ClientMessage) => bool {
     switch typeof(clientMsg) {
       case Header:
-        verifyHeader(clientMsg)
+        return verifyHeader(clientMsg)
       case Misbehaviour:
-        verifyHeader(clientMsg.h1)
-        verifyHeader(clientMsg.h2)
+        return verifyHeader(clientMsg.h1) && verifyHeader(clientMsg.h2)
     }
 }
 ```
 
 Verify validity of regular update to the Tendermint client
 
 ```typescript
-function verifyHeader(header: Header) {
+function verifyHeader(header: Header) => bool {
     clientState = get("clients/{header.identifier}/clientState")
-    // assert trusting period has not yet passed
-    assert(currentTimestamp() - clientState.latestTimestamp < clientState.trustingPeriod)
-    // assert header timestamp is less than trust period in the future. This should be resolved with an intermediate header.
-    assert(header.timestamp - clientState.latestTimeStamp < trustingPeriod)
+    // trusting period must have not yet passed
+    if clientState.latestTimestamp + clientState.trustingPeriod > currentTimestamp() {
```

Comment:
I wrote the conditions for the trusting period checks in a way that was easier for me to wrap my head around, but I would appreciate someone double checking they are fine.